crates/bevy_ecs/src/query/iter.rs

```diff
@@ -69,6 +69,68 @@ where
             current_index: 0,
         }
     }
+
+    /// Consumes `self` and returns true if there were any elements remaining in this iterator.
+    #[inline(always)]
+    pub(crate) fn any_remaining(mut self) -> bool {
+        // NOTE: this mimics the behavior of `QueryIter::next()`, expect that it
+        // never gets a `Self::Item`.
+        unsafe {
+            if self.is_dense {
+                loop {
+                    if self.current_index == self.current_len {
+                        let table_id = match self.table_id_iter.next() {
+                            Some(table_id) => table_id,
+                            None => return true,
+                        };
+                        let table = &self.tables[*table_id];
+                        self.fetch.set_table(&self.query_state.fetch_state, table);
```

Comment:
You can remove anything involving self.fetch from this impl.

Reply:
(However I assume that `self.filter` must be left in?)

Reply:
yup!

Reply:
Cool, PR was update accordingly :)